sdk/cosmos/src/operations/create_database.rs

```diff
@@ -3,44 +3,71 @@ use crate::prelude::*;
 use crate::resources::Database;
 use crate::ResourceQuota;
 use azure_core::headers::{etag_from_headers, session_token_from_headers};
-use azure_core::{collect_pinned_stream, Request as HttpRequest, Response as HttpResponse};
+use azure_core::{collect_pinned_stream, Context, PipelineContext, Response as HttpResponse};
 use chrono::{DateTime, Utc};
 
 #[derive(Debug, Clone)]
-pub struct CreateDatabaseOptions {
+pub struct CreateDatabaseBuilder {
+    client: CosmosClient,
+    database_name: String,
     consistency_level: Option<ConsistencyLevel>,
+    context: Context,
 }
 
-impl CreateDatabaseOptions {
-    pub fn new() -> Self {
+impl CreateDatabaseBuilder {
+    pub(crate) fn new(client: CosmosClient, database_name: String) -> Self {
         Self {
+            client,
+            database_name,
             consistency_level: None,
+            context: Context::new(),
         }
     }
 
     setters! {
         consistency_level: ConsistencyLevel => Some(consistency_level),
+        context: Context => context,
     }
-}
 
-impl CreateDatabaseOptions {
-    pub(crate) fn decorate_request(
-        &self,
-        request: &mut HttpRequest,
-        database_name: &str,
-    ) -> crate::Result<()> {
-        #[derive(Serialize)]
-        struct CreateDatabaseRequest<'a> {
-            pub id: &'a str,
-        }
-        let req = CreateDatabaseRequest { id: database_name };
+    pub fn insert<E: Send + Sync + 'static>(&mut self, entity: E) -> &mut Self {
+        self.context.insert(entity);
+        self
+    }
+
+    pub fn into_future(self) -> CreateDatabase {
```

Comment:
For the `context` method to work in a chaining fashion, this type should take `self` by-reference:
```suggestion
    pub fn into_future(&self) -> CreateDatabase {
```
This may require additional calls to `clone` in the body of `Box::pin`.